frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2212,6 +2212,21 @@ doIsCandidateApplicableInitial(Context* context,
     }
   }
 
+  if (ci.isMethodCall() && ci.name() == "init") {
+    // TODO: test when record has defaults for type/param fields
+    auto recv = ci.calledType();
+    auto fn = parsing::idToAst(context, candidateId)->toFunction();
```

Comment:
Needs an assert of check for if it's not a function, depending on if it's possible for it not to be a function.

Reply:
Ah i see it's below. Probably needs to be moved above the init handling.